unit_test/sparse/Test_Sparse_spmv_bsr.hpp

```diff
@@ -0,0 +1,770 @@
+
+    //
+    // Compute the reference product with a CrsMatrix variable
+    //
+    KokkosSparse::spmv(fOp, alpha, Acrs, xref, beta, ycrs);
+
+    block_vector_t ybsr("bsr_product_result", nRow, nrhs);
+    auto h_ybsr = Kokkos::create_mirror_view(ybsr);
+    for (int jc = 0; jc < nrhs; ++jc)
+      for (lno_t ir = 0; ir < nRow; ++ir) h_ybsr(ir, jc) = h_y0(ir, jc);
+    Kokkos::deep_copy(ybsr, h_ybsr);
+
+    // Create the BsrMatrix for the check test
+    KokkosSparse::Experimental::BsrMatrix<scalar_t, lno_t, device, void,
+                                          size_type>
+        Absr(Acrs, blockSize);
+
+    //
+    // Compute the product with the BsrMatrix format
+    //
+    KokkosSparse::spmv(fOp, alpha, Absr, xref, beta, ybsr);
+
+    Kokkos::deep_copy(h_ycrs, ycrs);
+    Kokkos::deep_copy(h_ybsr, ybsr);
+
+    //
+    // Compare the two products
+    //
+    double error = 0.0, maxNorm = 0.0;
+    for (int jc = 0; jc < nrhs; ++jc) {
+      for (int ir = 0; ir < nRow; ++ir) {
+        error   = std::max<double>(error, Kokkos::ArithTraits<scalar_t>::abs(
+                                            h_ycrs(ir, jc) - h_ybsr(ir, jc)));
+        maxNorm = std::max<double>(
+            maxNorm, Kokkos::ArithTraits<scalar_t>::abs(h_ycrs(ir, jc)));
+      }
+    }
+
+    double tmps =
+        static_cast<double>(Kokkos::ArithTraits<scalar_t>::abs(alpha)) +
+        static_cast<double>(Kokkos::ArithTraits<scalar_t>::abs(beta));
+    if ((tmps > 0.0) && (maxNorm == 0)) {
+      std::cout << " BSR - SpMV times MV >> blockSize " << blockSize
+                << " maxNorm " << maxNorm << " error " << error << " alpha "
+                << alpha << " beta " << beta << "\n";
+      num_errors += 1;
+    }
+
+    auto tol = ((nnz / nRow) + 1) *
+               static_cast<double>(Kokkos::ArithTraits<scalar_t>::abs(
+                   Kokkos::ArithTraits<scalar_t>::epsilon()));
+    if (error > tol * maxNorm) {
+      std::cout << " BSR - SpMV times MV >> blockSize " << blockSize
+                << " ratio " << error / maxNorm << " tol " << tol << " maxNorm "
+                << maxNorm << " alpha " << alpha << " beta " << beta << "\n";
+      num_errors += 1;
+    }
+
+  }  // for (int blockSize = 1; blockSize <= bMax; ++blockSize)
+}
+
+}  // namespace Test_Bsr
+
+template <typename scalar_t, typename lno_t, typename size_type,
+          typename device>
+void testSpMVBsrMatrix() {
+  //
+  // Test for the operation y <- alpha * Op(A) * x + beta * y
+  //
+
+  // Define the function Op: Op(A) = A, Op(A) = conj(A), Op(A) = A^T, Op(A) =
+  // A^H
+  std::vector<char> modes = {'N', 'C', 'T', 'H'};
+
+  // Define a set of pairs (alpha, beta)
+  std::vector<double> testAlphaBeta = {0.0, 0.0, -1.0, 0.0,
+                                       0.0, 1.0, 3.1,  -2.5};
+
+  //
+  // Set the largest block size for the block matrix
+  // The code will create matrices with block sizes 1, .., bMax
+  //
+  const lno_t bMax = 13;
```

Comment:
This could be a `constexpr`.
Also in general we want to check a few corner cases in term of matrix dimensions:
1. zero rows in the matrix, this can happen when an MPI rank does not own local data
2. zero columns, this is more rare but we should check just in case

Reply:
Done.